pkg/gce-pd-csi-driver/node_test.go

```diff
@@ -328,6 +341,87 @@ func TestNodeStageVolume(t *testing.T) {
 	}
 }
 
+func TestNodeExpandVolume(t *testing.T) {
+	// TODO: Add tests/functionality for non-existant volume
+	var resizedBytes int64 = 2000000000
+	volumeID := "project/test001/zones/c1/disks/testDisk"
+	testCases := []struct {
+		name         string
+		req          *csi.NodeExpandVolumeRequest
+		blockDevice  bool
+		expRespBytes int64
+		expErrCode   codes.Code
+	}{
+		{
+			name: "ext4 fs expand",
+			req: &csi.NodeExpandVolumeRequest{
+				VolumeId:   volumeID,
+				VolumePath: "some-path",
+				CapacityRange: &csi.CapacityRange{
+					RequiredBytes: resizedBytes,
+				},
+			},
+			blockDevice:  false,
+			expRespBytes: resizedBytes,
+		},
+		{
+			name: "block device expand",
+			req: &csi.NodeExpandVolumeRequest{
+				VolumeId:   volumeID,
+				VolumePath: "some-path",
+				CapacityRange: &csi.CapacityRange{
+					RequiredBytes: resizedBytes,
+				},
+			},
+			blockDevice: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Logf("Test case: %s", tc.name)
+
+		execCallback := func(cmd string, args ...string) ([]byte, error) {
+			if cmd == "blkid" {
+				if tc.blockDevice {
+					// blkid returns exit code 2 when run on unformatted device
+					return nil, utilexec.CodeExitError{
+						Err:  errors.New("this is an exit error"),
+						Code: 2,
```

Comment:
hopefully it doesn't overload 2 with any other error

Reply:
> If the specified token was not found, or no (specified) devices could be identified, an exit code of 2 is returned. 

we search specifically for tokens `TYPE` and `PTTYPE`. I guess this error could also occur for any other reason the tokens don't exist (of which I can't think of any currently).

> For usage or other errors, an exit code of 4 is returned.